python/rmm/tests/test_rmm.py

```diff
@@ -942,3 +943,73 @@ def test_rmm_device_buffer_copy(cuda_ary, make_copy):
     result = db_copy.copy_to_host()
 
     np.testing.assert_equal(expected, result)
+
+
+@pytest.mark.parametrize("level", rmm.logging_level)
+def test_valid_logging_level(level):
+    warnings.filterwarnings(
+        "ignore", message="RMM will not log warning level TRACE."
+    )
+    warnings.filterwarnings(
+        "ignore", message="RMM will not log warning level DEBUG."
+    )
+    rmm.set_logging_level(level)
+    assert rmm.get_logging_level() == level
+    # Also test setting using any value or name in the enum
+    rmm.set_logging_level(level.value)
+    assert rmm.get_logging_level() == level
+    rmm.set_logging_level(level.name)
+    assert rmm.get_logging_level() == level
+    # Test case insensitive
+    rmm.set_logging_level(level.name.title())
+    assert rmm.get_logging_level() == level
+    # reset to default
+    rmm.set_logging_level("INFO")
+
+
+def test_invalid_logging_level():
+    with pytest.raises(KeyError):
+        rmm.set_logging_level("invalid")
+    with pytest.raises(ValueError):
+        rmm.set_logging_level(100)
+    with pytest.raises(ValueError):
+        rmm.set_logging_level(None)
+    with pytest.raises(ValueError):
+        rmm.set_logging_level(1.2345)
+    with pytest.raises(ValueError):
+        rmm.set_logging_level([1, 2, 3])
+
+
+@pytest.mark.parametrize("level", rmm.logging_level)
+def test_valid_logging_flush_level(level):
+    warnings.filterwarnings(
+        "ignore", message="RMM will not log warning level TRACE."
+    )
+    warnings.filterwarnings(
+        "ignore", message="RMM will not log warning level DEBUG."
+    )
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**nitpick (non-blocking):** ‏As above, if the proposed warning message change is preferred
```suggestion
    warnings.filterwarnings(
        "ignore", message="RMM will not log at level TRACE."
    )
    warnings.filterwarnings(
        "ignore", message="RMM will not log at level DEBUG."
    )
```

Reply:
And, as above, we want to (either way) wrap this test body in a `with warnings.catch_warnings():` block